components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1155,38 +1046,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 }
             }
 
-            let project: Project | undefined = undefined;
-            if (options.projectId) {
-                project = await this.projectsService.getProject(user.id, options.projectId);
-            } else if (CommitContext.is(context)) {
-                const projects = await this.projectsService.findProjectsByCloneUrl(
-                    user.id,
-                    context.repository.cloneUrl,
-                );
-                if (projects.length > 1) {
-                    throw new ApplicationError(ErrorCodes.BAD_REQUEST, "Multiple projects found for clone URL.");
-                }
-                if (projects.length === 1) {
-                    project = projects[0];
-                }
-            }
-
-            const mayStartWorkspacePromise = this.workspaceService.mayStartWorkspace(
-                ctx,
-                user,
-                options.organizationId,
-                runningInstancesPromise,
-            );
-
-            const prebuiltWorkspace = project?.settings?.prebuilds?.enable
-                ? await this.findPrebuiltWorkspace(ctx, user, project.id, context, options.organizationId)
-                : undefined;
-            if (WorkspaceContext.is(prebuiltWorkspace)) {
-                ctx.span?.log({ prebuild: "available" });
-                context = prebuiltWorkspace;
-            }
-
-            await mayStartWorkspacePromise;
+            await this.workspaceService.mayStartWorkspace(ctx, user, options.organizationId, runningInstancesPromise);
```

Comment:
Corollary to [this comment](https://github.com/gitpod-io/gitpod/pull/19076/files#r1400241031): I think the call should go into WorkspaceService.createWorkspace. Then we could make the method private as well I think!